### ISSUE TITLE
Links getting cut off.

### ISSUE THREAD

**Issue by marshalmiller** [marshalmiller/linkrot#30]
When collecting links it does not recognize links that span two lines. It will cut off the link.

**Comment by IanFindlay** [comment]
Stumbled upon this issue and had the immediate thought of removing all of the spaces in the text as part of the [extract_urls function](https://github.com/marshalmiller/linkrot/blob/55ee1c7a8ed89024bd8872a19127e7b8e83a4612/linkrot/extractor.py#L25-L33). Not sure how this would perform on really large files or if it addresses the type of cutting off that you are concerned about.

If you think that would be an adequate solution to the issue I am more than happy to make the changes and make a PR.

**Comment by veracioux** [comment]
The way I understand it, only newlines should be ignored, not all spaces.

@marshalmiller Is this the behavior you are looking for?

From this text
```
Download the app at https://download.com.
Contribute on https://github.com
/marshalmiller/linkrot.
See htt
ps://example.org/README.md for details. Post issues and PRs on:
https://example.org/issues
https://example.org/pulls
```
to extract the following URLs:
- `https://download.com`
- `https://github.com/marshalmiller/linkrot`
- `https://example.org/README.md`
- `https://example.org/issues`
- `https://example.org/pulls`

The problem with this is that by simply ignoring newlines we will get `https://github.com/marshalmiller/linkrot.See` and `https://example.org/issueshttps://example.org/pulls` as extracted URLs.

Using the first example, maybe we should extract both `https://github.com/marshalmiller/linkrot` and `https://github.com/marshalmiller/linkrot.See`, and discard the one that is broken.

Just laying out my thoughts. I even started implementing this only to realize it is more complicated than what I thought initially :)

**Comment by marshalmiller** [comment]
Yeah. I'm not sure if removing all of the spaces is the solution. I will run some tests tomorrow and see if I can break something.

**Comment by marshalmiller** [comment]
Addressed in Version 3.0
